Lib/idlelib/pyshell.py

```diff
@@ -635,6 +635,9 @@ def execfile(self, filename, source=None):
         if source is None:
             with tokenize.open(filename) as fp:
                 source = fp.read()
+                if use_subprocess:
+                    source = (f"__file__ = r'{os.path.abspath(filename)}'\n" +
+                              source + "\ndel __file__")
```

Comment:
This gives a SyntaxError if there's an unescaped quote in the path or file name (such as `/home/cheryl's/tests/`).

Using triple quotes works:
```f"__file__ = r'''{os.path.abspath(filename)}'''\n"```

Reply:
Great catch.  Changed.